django/middleware/cache.py

```diff
@@ -171,6 +174,17 @@ def process_request(self, request):
             request._cache_update_cache = True
             return None  # No cache information available, need to rebuild.
 
+        # Derive the age estimation of the cached entry using max_age and the Expires
+        # timestamp. max_age_seconds is None if CACHE_MIDDLEWARE_SECONDS is
+        # misconfigured to be a float.
+        max_age_seconds = get_max_age(response)
+        if max_age_seconds is not None:
+            expires_timestamp = parse_http_date(response["Expires"])
```

Comment:
🤔 I'm wondering out loud, mostly. Is this a scenario where we care/need to handle the possibility of an exception being raised? There's other places which act similarly upon "response" headers[^1] and do so by leveraging `parse_http_date_safe`[^2] to avoid crashing, but it's decidedly unclear to me as to whether that's relevant here...

Just flagging it for consideration, as such.

[^1]: "last modified" & "if modified since"
[^2]: which just catches and returns `None`

Reply:
I thought about that too, but came to the conclusion, that extra caution is not required here, since the `Expires` response header is set by a django cache. 

`response.headers["Expires"] = http_date(time.time() + cache_timeout)`

Two cases you mentioned in `cache.py` are dealing with request headers `If-Modified-Since` and `If-Unodified-Since` that could likely contain badly formatted date strings and need `parse_http_date_safe`

`Last-Modified` from a request will be saved in the cache as is and could be badly formatted as well and is used for a conditional response.

Reply:
> ...since the `Expires` response header is set by a django cache.

Yes, but only if it's not already set.

Reply:
> > ...since the `Expires` response header is set by a django cache.
> 
> Yes, but only if it's not already set.

True. I missed `Expires` being part of `_not_modified`.

Reply:
> What do you think about the following change?

Looks good. Made the change.